recirq/time_crystals/dtctask.py

```diff
@@ -0,0 +1,254 @@
+
+
+@recirq.json_serializable_dataclass(namespace='recirq.readout_scan',
+                                    registry=recirq.Registry,
+                                    frozen=False)
+class DTCTask:
+    """ A task for managing inputs to a Discrete Time Crystal experiment, over some number of disorder instances
+
+    Attributes:
+        dataset_id: unique identifier for this dataset
+        qubits: a chain of connected qubits available for the circuit
+        disorder_instances: number of disorder instances averaged over
+        initial_states: initial state of the system used in circuit
+        g: thermalization constant used in circuit
+        local_fields: random noise used in circuit
+        thetas: theta parameters for FSim Gate used in circuit
+        zetas: zeta parameters for FSim Gate used in circuit
+        chis: chi parameters for FSim Gate used in circuit
+        phis: phi parameters for FSim Gate used in circuit
+        gammas: gamma parameters for FSim Gate used in circuit
+
+    """
+    # Task parameters
+    dataset_id: str
+
+    # experiment parameters
+    qubits: Sequence[cirq.Qid]
+    disorder_instances: int
+
+    # FSim Gate parameters
+    # ndarrays in this section are in shape (disorder_instances, len(qubits) - 1)
+    g: int
+    initial_states: np.ndarray
+    local_fields: np.ndarray
+
+    # FSim Gate Parameters
+    # ndarrays in this section are in shape (disorder_instances, len(qubits) - 1)
+    thetas: np.ndarray
+    zetas: np.ndarray
+    chis: np.ndarray
+    gammas: np.ndarray
+    phis: np.ndarray
+
+
+    def __init__(
+            self,
+            qubits: Optional[Sequence[cirq.Qid]] = None,
+            disorder_instances: Optional[int] = None,
+            g: Optional[int] = None,
+            initial_state: Optional[np.ndarray] = None,
+            initial_states: Optional[np.ndarray] = None,
+            local_fields: Optional[np.ndarray] = None,
+            thetas: Optional[np.ndarray] = None,
+            zetas: Optional[np.ndarray] = None,
+            chis: Optional[np.ndarray] = None,
+            gammas: Optional[np.ndarray] = None,
+            phis: Optional[np.ndarray] = None
+            ):
+
+        self.dataset_id = datetime.datetime.utcnow()
+
+        self.disorder_instances = 36 if disorder_instances is None else disorder_instances
+
+        self.g = 0.94 if g is None else g
+
+        if qubits is None:
+            qubit_locations = [(3, 9), (3, 8), (3, 7), (4, 7), (4, 8), (5, 8), (5, 7), (5, 6), (6, 6), (6, 5), (7, 5), (8, 5), (8, 4), (8, 3), (7, 3), (6, 3)]
+            self.qubits = [cirq.GridQubit(*idx) for idx in qubit_locations]
+        else:
+            self.qubits = qubits
+
+        num_qubits = len(self.qubits)
+
+        # only enable use of initial_state or initial_states
+        assert initial_state is None or initial_states is None, 'do not supply both initial_state and initial_states'
```

Comment:
this seems inappropriate. Why have both as names? You really shouldn't have the signature for `__init__` in a dataclass differ from the list of fields